src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseCountProperly.cs

```diff
@@ -0,0 +1,679 @@
+        {
+            if (!IsIntEqualsMethod(parent))
+            {
+                return;
+            }
+
+            IOperation constantOperation = isInstance ? parent.Arguments[0].Value : parent.Instance;
+            if (!TryGetZeroOrOneConstant(constantOperation, out int constant) || constant != 0)
+            {
+                // Parent expression doesn't met the criteria to be replaced; fallback on analysis for standalone call to Count().
+                if (!isAsync && !hasPredicate && operation.Kind == OperationKind.Invocation)
+                {
+                    AnalyzeCountInvocationOperation(context, (IInvocationOperation)operation);
+                }
+                return;
+            }
+
+            if (allowLinq && isAsync)
+            {
+                ReportCA1828(context, shouldNegate: true, isInstance ? OperationEqualsInstance : OperationEqualsArgument, methodName!, parent);
+            }
+            else
+            {
+                ITypeSymbol? type = operation.GetInstanceType();
+                if (type != null)
+                {
+                    // If the invocation has a predicate we can only suggest CA1827; 
+                    // otherwise, we would loose the lambda if we suggest CA1829 or CA1836. 
+                    if (hasPredicate)
+                    {
+                        ReportCA1827(context, shouldNegate: true, isInstance ? OperationEqualsInstance : OperationEqualsArgument, methodName!, parent);
+                    }
+                    else
+                    {
+                        if (TypeContainsVisibileProperty(context, type, IsEmpty, SpecialType.System_Boolean))
+                        {
+                            ReportCA1836(context, !isInstance, shouldNegate: false, parent);
+                        }
+                        else if (TypeContainsVisibileProperty(context, type, Length, SpecialType.System_Int32, SpecialType.System_UInt64))
+                        {
+                            ReportCA1829(context, Length, operation);
+                        }
+                        else if (TypeContainsVisibileProperty(context, type, Count, SpecialType.System_Int32, SpecialType.System_UInt64))
+                        {
+                            ReportCA1829(context, Count, operation);
+                        }
+                        else if (allowLinq)
+                        {
+                            ReportCA1827(context, shouldNegate: true, isInstance ? OperationEqualsInstance : OperationEqualsArgument, methodName!, parent);
+                        }
+                    }
+                }
+            }
+        }
+
+        private static void AnalyzeCountInvocationOperation(OperationAnalysisContext context, IInvocationOperation invocationOperation)
+        {
+            ITypeSymbol? type = invocationOperation.GetInstanceType();
```

Comment:
Should we just bail out here if type is null?

Reply:
That's a good point, I don't think so, I think we could find some other way to get `type` when there is no instance, but as I said on https://github.com/dotnet/roslyn-analyzers/pull/3584#discussion_r427486672, I haven't been able to get null, even with static members.

Reply:
OK, I was able to get null from GetInstanceType with the following snippet:
```cs
public class Test
{
    public static int Count {get;}
    public bool IsEmpty;
    public bool Foo => Count > 0;
}
```

So, we would be bailing just with static properties and static methods, static extension methods does still have an instance so those are fine.

Is it OK to not report on that case?

Reply:
I think it seems fine to bail out on static cases.